File: 0241-different-ways-to-add-parentheses/0241-different-ways-to-add-parentheses.java
class Solution {
    public List<Integer> diffWaysToCompute(String expression) {
        List<Integer> list = new ArrayList<>();
        for(int i=0;i<expression.length();++i){
            char oper = expression.charAt(i);
            if(oper == '+' || oper == '-' || oper == '*'){
                List<Integer> s1 = diffWaysToCompute(expression.substring(0,i));
                List<Integer> s2 = diffWaysToCompute(expression.substring(i+1));
                for(int a: s1){
                    for(int b: s2){
                        if(oper == '+') list.add(a+b);
                        else if(oper == '-') list.add(a-b);
                        else if(oper == '*') list.add(a*b);
                    }
                }
            }
        }
        if(list.isEmpty()){
            list.add(Integer.parseInt(expression));
        }
        return list;
    }
}

File: 0241-different-ways-to-add-parentheses/0241-different-ways-to-add-parentheses.js
/**
 * @param {string} expression
 * @return {number[]}
 */
var diffWaysToCompute = function(expression) {
    let arr = [];
    for(let i=0;i<expression.length;i++){
        let oper = expression.charAt(i);
        if(oper === '+' || oper === '-' || oper === '*'){
            let s1 = diffWaysToCompute(expression.substring(0,i));
            let s2 = diffWaysToCompute(expression.substring(i+1));
            for(let a of s1){
                for(let b of s2){
                    if(oper === '+'){
                        arr.push(a+b);
                    }else if(oper === '-'){
                        arr.push(a-b);
                    }else if(oper === '*'){
                        arr.push(a*b);
                    }
                }
            }
        }
    }
    if(arr.length === 0){
        arr.push(parseInt(expression));
    }
    return arr;
};